src/core/thread/mle_router.cpp

```diff
@@ -2540,7 +2540,7 @@ void MleRouter::HandleChildUpdateRequest(const Message &         aMessage,
         {
             child->SetCslChannel(static_cast<uint8_t>(cslChannel.GetChannel()));
         }
-        else
+        else if (child->GetCslChannel() == 0)
```

Comment:
I think it's allowed for a child to change and update its CSL channel. I didn't find descriptions in spec to forbid so. What's the reason doing so?